### PR TITLE
CNF-15499: Allow references to supply per-template description text

### PR DESCRIPTION
**Pull request by lack**
Signed-off-by: Jim Ramsay <jramsay@redhat.com>

### CODE REVIEW

File: docs/reference-config-guide-v2.md [hunk 1]
```diff
@@ -88,6 +88,32 @@ components:
...
+which can contain instructions or a URL referencing extra
+documentation about the template in question, such as why it
+is required or which fields are optional.
+
+```yaml
+components:
+  - name: ExampleComponent1
+    allOf:
+      - path: RequiredTemplate1.yaml
+        helpText: |-
```
Comment (natifridman) [inline docs/reference-config-guide-v2.md:108]: Should we call it `description` instead?

File: pkg/compare/referenceV1.go [hunk 2]
```diff
@@ -204,6 +214,7 @@ func (config ReferenceTemplateConfigV1) GetFieldsToOmitRefs() []string {
 type ReferenceTemplateV1 struct {
 	*template.Template `json:"-"`
 	Path               string                    `json:"path"`
+	Description        string                    `json:"description,omitempty"`
```
Comment (nocturnalastro) [inline pkg/compare/referenceV1.go:217]: I think this should be on ReferenceTemplateV2. And the V1 template should just return an empty string.

Comment (lack) [reply pkg/compare/referenceV1.go:217]: I think it's "safe" to add new optional fields to older APIs though, and V1 could benefit from this as well at no extra cost.

Comment (nocturnalastro) [reply pkg/compare/referenceV1.go:217]: Yeah shouldn't be an issue I suppose. 
Could you add a test for it then just to make sure we don't break `getMissingCRs` for anyone who decided to go with V1 and use the new field.

Comment (lack) [reply pkg/compare/referenceV1.go:217]: Suddenly "no extra cost" became "adding another test", so I'm now less convinced ;)

**Comment by lack** [comment]
/override ci/prow/unit

The intermittent failure here is not related to this change.

File: pkg/compare/referenceV1.go (hunk 2 above)
Comment (lack) [reply pkg/compare/referenceV1.go:217]: Test added.

**Review by natifridman — APPROVED** [review]
/lgtm

**Pull request merged**